### ISSUE TITLE
Storyboard samples don't stop if skipped past end

### ISSUE THREAD

**Issue by bdach** [ppy/osu#11421]
Spotted when double-checking #11418. Opening so I don't forget about it. Tried a quick fix but that failed, so it needs proper looking at.

Reproduction steps:

1. Import and start https://osu.ppy.sh/beatmapsets/5381#osu/26547 (with storyboard enabled).
2. Skip once (to song start) while the intro SFX is still playing (`sfx/intro.mp3` in beatmap files).
3. The intro SFX will continue to play along with the song starting, while it stops on stable (leaving only the song).

Not sure how stable handles this, but it seems to.

**Comment by peppy** [comment]
Stable just halts all sample playback on skip (game-wide), for reference.

**Comment by Mysfit** [comment]
I looked into this for a bit, and it seems that there is also an issue where the storyboard SFX doesn't pause when the beatmap is paused. I'm adding this here instead of making a new issue because I suspect they are related, but I can create a new issue if needed. 

Reproduction steps:
  1. Import and start beatmap linked above with storyboard enabled.
  2. Once intro SFX starts playing (`sfx/intro.mp3`), pause the beatmap.
  3. The intro SFX will continue to play while the map is paused. 

I confirmed that stable does pause the SFX if the beatmap is paused.
